edb/graphql/translator.py

```diff
@@ -204,18 +229,25 @@ def visit_DocumentNode(self, node):
                 raise errors.QueryError(
                     f'unknown operation named "{operation_name}"')
 
-        stmt = translated[operation_name].stmt
-        for el in stmt.result.elements:
+        operation = translated[operation_name]
+        for el in operation.stmt.result.elements:
             # swap in the json bits
             if (isinstance(el.compexpr, qlast.FunctionCall) and
                     el.compexpr.func == 'to_json'):
 
                 # An introspection query; let graphql evaluate it for us.
+
+                vars = BookkeepDict(self._context.variables)
                 result = graphql.execute(
                     self._context.gqlcore.graphql_schema,
                     self._context.document_ast,
                     operation_name=operation_name,
-                    variable_values=self._context.variables)
+                    variable_values=vars)
+                for var in vars.touched:
+                    self._context.vars[var] = (
+                        self._context.vars.get(var)
+                        ._replace(critical=True)
```

Comment:
For better readability

```suggestion
                        self._context.vars.get(var)._replace(critical=True)
```

Reply:
Well, The first doesn't fit 79 chars as far as I remember. I don't find second more readable (the method name and argument name are much farther from each other).

Reply:
Changed to temporary var.

Reply:
> I don't find second more readable (the method name and argument name are much farther from each other).

FWIW it's the current code style we're using everywhere for code like this.